src/disk.c

```diff
@@ -814,15 +830,15 @@ static int disk_read(void) {
       assert(numfields >= 14);
 
       /* Kernel 4.18+, Discards */
-      if (numfields >= 18) {
+      if (numfields >= 18 && report_discard) {
```

Comment:
Please handle the config option just before calling `disk_submit_single` and always populate the `discard_*` and `flush_*` variables. That means line 1000 would look something like this:

```c
if (report_discard && ds->has_discard) {
```

Dito for `report_flush`.